daemon/logger/jsonfilelog/read.go

```diff
@@ -53,19 +55,47 @@ func (l *JSONFileLogger) readLogs(logWatcher *logger.LogWatcher, config logger.R
 	l.mu.Lock()
 
 	pth := l.writer.LogPath()
-	var files []io.ReadSeeker
+	var files []io.ReadCloser
+	var readSeekers []io.ReadSeeker
 	for i := l.writer.MaxFiles(); i > 1; i-- {
 		f, err := os.Open(fmt.Sprintf("%s.%d", pth, i-1))
 		if err != nil {
 			if !os.IsNotExist(err) {
 				logWatcher.Err <- err
 				break
 			}
+
+			cf, err := os.Open(fmt.Sprintf("%s.%d.gz", pth, i-1))
+			if err != nil {
+				if !os.IsNotExist(err) {
+					logWatcher.Err <- err
+					break
+				}
+				continue
+			}
+			files = append(files, cf)
+
+			rc, err := archive.DecompressStream(cf)
+			if err != nil {
+				logWatcher.Err <- err
+				break
+			}
+			defer rc.Close()
+
+			fileData, err := ioutil.ReadAll(rc)
```

Comment:
@LK4D4 @cpuguy83 I have implemented a seeker for a compressed stream. PTAL, 
If there is anything wrong, please point out the right direction, thanks so much!